newIDE/app/src/SceneEditor/InstancesSelection.js

```diff
@@ -24,21 +24,33 @@ export default class InstancesSelection {
     this.selection.length = 0;
   }
 
-  selectInstance(instance: gdInitialInstance, multiselect: boolean = false) {
+  selectInstance(
+    instance: gdInitialInstance,
+    multiselect: boolean = false,
+    layersVisibility: object
+  ) {
     if (this.isInstanceSelected(instance)) {
       if (multiselect) this.unselectInstance(instance);
 
       return;
     }
 
     if (!multiselect) this.clearSelection();
-    this.selection.push(instance);
+    if (layersVisibility[instance.getLayer()]) {
+      this.selection.push(instance);
+    }
   }
 
-  selectInstances(instances: [gdInitialInstance], multiselect: boolean) {
+  selectInstances(
+    instances: [gdInitialInstance],
+    multiselect: boolean,
+    layersVisibility: object
```

Comment:
`object` is not valid in Flow, it's `Object`. But we should almost always avoid using it. The proper typing here is `{[string]: boolean}` (a map of strings to booleans) :) 
(More info here: https://flow.org/en/docs/types/objects/#toc-objects-as-maps)

Reply:
Well, it's `?{[string]: boolean}` if null/undefined is supported :)